src/graphiql-explain/useExplain.js

```diff
@@ -3,16 +3,87 @@ import { explainDataManager } from './ExplainDataManager'
 
 export const useExplain = () => {
   const [explain, setExplain] = useState(explainDataManager.getExplainData())
+  const [defaultExplain, setDefaultExplain] = useState(
+    explainDataManager.getExplainData()
+  )
+  const [pathOrder, setPathOrder] = useState(0)
+  const [timeOrder, setTimeOrder] = useState(0)
+
   useEffect(() => {
     const eventListener = explainDataManager.addEventListener(
       'updateExplainData',
       (e, value) => {
+        setDefaultExplain(() => e.target.explainData || [])
         setExplain(_ => e.target.explainData || [])
       }
     )
     return () => {
       explainDataManager.removeEventListener('updateExplainData', eventListener)
     }
   }, [])
-  return { explain }
+
+  const searchByField = fieldName => {
+    return event => {
+      setExplain(_ => {
+        return defaultExplain.filter(element => {
+          return element[fieldName].includes(event.target.value)
+        })
+      })
+    }
+  }
+
+  const sort = (fieldName = 'path') => {
+    return order => {
+      let localOrder = order
+      if (!order || order === 0) {
+        return explain
+      } else if (order >= 1) {
+        localOrder = 1
+      } else {
+        localOrder = -1
+      }
+      setExplain(_ => {
+        return explain.sort((a, b) => {
+          return localOrder * `${a[fieldName]}`.localeCompare(`${b[fieldName]}`)
+        })
+      })
+    }
+  }
+
+  const sortPath = sort('path')
+  const sortTime = sort('time')
+
+  const changePathOrder = () => {
```

Comment:
Add this in a useCalback. It's not currently strictly required but it's exported and if some component uses it will create an infinite loop.